packages/@lwc/ssr-compiler/src/compile-template/component.ts

```diff
@@ -52,7 +52,11 @@ function getChildAttrsOrProps(
                     : attr.value.value;
             return b.property('init', key, b.literal(value));
         } else if (attr.value.type === 'Literal' && typeof attr.value.value === 'boolean') {
-            return b.property('init', key, b.literal(attr.value.value));
+            return b.property(
+                'init',
+                key,
+                b.literal(attr.type === 'Attribute' ? '' : attr.value.value)
+            );
```

Comment:
This will convert both `true` and `false` to `''`; is that desired?

Reply:
You're right, I thought these attributes were from the template, but they also include attributes of reflected props.

Reply:
![Screenshot 2024-10-04 at 11 16 41](https://github.com/user-attachments/assets/5466b16a-6623-4174-ac7a-2154a32a338f)
Actually, it looks like when you set an attribute with an expression (the only way a boolean attribute can be `false`), it drops through to the `else` because `attr.value.type === 'Identifier'`.